tests/unit_tests/udf_wrapper_params/question_answering/multiple_locations_multiple_batch_complete.py

```diff
@@ -0,0 +1,59 @@
+from tests.unit_tests.udf_wrapper_params.question_answering.\
+    mock_question_answering import \
+    MockQuestionAnsweringFactory, MockQuestionAnsweringModel, MockPipeline
+
+
+def udf_wrapper():
+    from exasol_udf_mock_python.udf_context import UDFContext
+    from exasol_transformers_extension.udfs.question_answering_udf import \
+        QuestionAnswering
+    from tests.unit_tests.udf_wrapper_params.question_answering. \
+        mock_sequence_tokenizer import MockSequenceTokenizer
+    from tests.unit_tests.udf_wrapper_params.question_answering.\
+        multiple_locations_multiple_batch_complete \
+        import MultipleModelLocationsMultipleBatchComplete \
+        as params
+
+    udf = QuestionAnswering(
+        exa,
+        batch_size=params.batch_size,
+        pipeline=params.mock_pipeline,
+        base_model=params.mock_factory,
+        tokenizer=MockSequenceTokenizer)
+
+    def run(ctx: UDFContext):
+        udf.run(ctx)
+
+
+class MultipleModelLocationsMultipleBatchComplete:
+    """
+    multiple locations, multiple batch, last batch complete
+    """
+    batch_size = 2
+    data_size = 2
+
+    input_data = [(None, "bfs_conn1", "sub_dir1", "model1",
+                   "question", "context")] * data_size + \
+                 [(None, "bfs_conn2", "sub_dir1", "model1",
+                   "question", "context")] * data_size + \
+                 [(None, "bfs_conn1", "sub_dir2", "model1",
+                   "question", "context")] * data_size + \
+                 [(None, "bfs_conn1", "sub_dir1", "model2",
+                   "question", "context")] * data_size
+    output_data = [("bfs_conn1", "sub_dir1", "model1", "question",
+                    "context", "answer 1", 0.1)] * data_size + \
+                  [("bfs_conn2", "sub_dir1", "model1", "question",
+                    "context", "answer 1", 0.1)] * data_size + \
+                  [("bfs_conn1", "sub_dir2", "model1", "question",
+                    "context", "answer 1", 0.1)] * data_size + \
+                  [("bfs_conn1", "sub_dir1", "model2", "question",
+                    "context", "answer 2", 0.2)] * data_size
+
+    mock_factory = MockQuestionAnsweringFactory({
```

Comment:
I think, we need a different factory here, because we need to check, if it uses the correct model accordingly the bucketfs_con, sub_dir and model. So, we need to return a different model (with different answer and score) for each combination of bucketfs_con, sub_dir and model

Applies also to all other MultiLocationTests

Reply:
We can do it (different model for each combination) by keeping the current Factory class as it is. We just need to udpate the inputs. e.g.: 
```
    mock_factory = MockQuestionAnsweringFactory({
        "model1.1": MockQuestionAnsweringModel(answer="answer 1", score=0.1),
        "model1.2": MockQuestionAnsweringModel(answer="answer 2", score=0.2),
        ..
    })
```

Reply:
The important part to test is, same model name, but different location. See, https://github.com/exasol/transformers-extension/pull/23#discussion_r936504276